src/main/java/core/mate/academy/service/TrackMachineProducerImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TrackMachineProducerImpl implements MachineProducer {
```

Comment:
let's parameterize MachineProducer with Track here, that will put obligation to parameterize get method with right type
and same in each impl